modules/post/multi/gather/memory_search.rb

```diff
@@ -0,0 +1,99 @@
+# -*- coding: binary -*-
+
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::MemorySearch
+
+  def initialize(info = {})
+    memory_search_info = initialize_memory_search(
+      info, platforms: %w[linux unix osx windows]
+    )
+
+    super(
+      update_info(
+        memory_search_info,
+        {}
+      )
+    )
+
+    register_options(
+      [
+        ::Msf::OptString.new('PROCESS_NAMES_GLOB', [false, 'Regular expression used to target processes', 'ssh.*']),
```

Comment:
glob != regex

https://en.wikipedia.org/wiki/Glob_(programming)
https://en.wikipedia.org/wiki/Regular_expression

We should ensure we're using the right terminology here

Regex has a downside here of the user providing `ssh.exe`, and it could match something unintentionally with the `.` directive matching any character rather than the ascii period that the user might have expected